nncf/experimental/tensorflow/quantization/quantizers.py

```diff
@@ -0,0 +1,310 @@
+        """
+        return self._signed_var.numpy() < 0.0
+
+    def build(self, nncf_network) -> None:
+        shape = None
+        if self.per_channel:
+            self.setup_input_transformation()
+            shape = (_get_channel_size(self._input_shape, self.channel_axes),)
+
+        prefix = self.name.replace('/', '^')
+
+        self._scale_var = nncf_network.add_weight(
+            f'{prefix}^scale',
+            shape=shape,
+            initializer=tf.keras.initializers.Constant(1.0),
+            trainable=True
+        )
+
+        self._signed_var = nncf_network.add_weight(
+            f'{prefix}^signed',
+            initializer=tf.keras.initializers.Constant(
+                -1.0 if self.signedness_to_force in (True, None) else 0.0
+            ),
+            trainable=False
+        )
+
+    def apply_overflow_fix(self):
+        if self.num_bits != 8 or not self.half_range:
+            raise RuntimeError('Attempt to apply saturation issue fix '
+                               'to quantizer which is not configured for that.')
+
+        # Multiplier to expand scale from 7 bit to 8 bit
+        multiplier = 127 / 63 if self.narrow_range else 255 / 127
+        self._scale_var.assign(multiplier * self._scale_var)
+        self._eps *= multiplier
+        self._half_range = False
+
+    def quantize(self, inputs):
+        num_bits = self.num_bits - 1 if self.half_range else self.num_bits
+        return symmetric_quantize(
+            inputs,
+            self._scale_var,
+            self._signed_var,
+            num_bits,
+            self.per_channel,
+            self.narrow_range,
+            self._eps
+        )
+
+    def apply_range_initialization(self, min_values, max_values, min_range=0.1, eps=0.01):
+        signed, scale = symmetric_range_initialization(
+            min_values, max_values, min_range, eps, self.signedness_to_force
+        )
+        self._signed_var.assign(signed)
+        self._scale_var.assign(scale)
+
+
+@NNCF_QUANTIZATION_OPERATIONS_V2.register(QuantizationMode.ASYMMETRIC)
+class AsymmetricQuantizerV2(QuantizerV2):
+    """
+    Represents the nncf operation that performs asymmetric quantization.
+    """
+
+    def __init__(self,
+                 name: str,
+                 qspec: TFQuantizerSpec,
+                 input_type: InputType,
+                 input_shape: Optional[List[int]] = None,
+                 channel_axes: Optional[List[int]] = None):
+        """
+        Initializes the internal state of the symmetric quantizer.
+        """
+        super().__init__(name, qspec, input_type, input_shape, channel_axes)
+        # Specification of the quantization
+        self._signedness_to_force = None
+        # Following variables are initialized inside the `build()` method.
+        self._input_low_var = None  # type: tf.Variable
+        self._input_range_var = None  # type: tf.Variable
+
+    @property
+    def mode(self) -> str:
```

Comment:
Type hint is invalid, should return `QuantizationMode`

Reply:
Thanks for the comment! It doesn't seem like my mistake. The problem is in the `QuantizationMode` class. Please see the version from the develop branch below
https://github.com/openvinotoolkit/nncf/blob/95070edfdbe2c682be24b47c4bdf2038fddd1487/nncf/common/quantization/structs.py#L21-L25

Thus,
```python
 isinstance(QuantizationMode.SYMMETRIC, QuantizationMode)
```
is `False`, but
```python
 isinstance(QuantizationMode.SYMMETRIC, str)
```
is `True`. It is for this reason I used `str` here.

Reply:
I created the ticket (Ref: 79918) to improve the situation.